src/paths/steiner.cpp

```diff
@@ -0,0 +1,574 @@
+
+                        int_set DMinusE = D;
+
+                        /*
+                         *  A set with Singleton value E removed from subset D
+                         */
+                        for (auto iter :  DMinusE) {
+                            if (iter == E) {
+                                iter = DMinusE.erase(iter);
+                                break;
+                            }
+                        }
+
+                        igraph_integer_t indexOfSubsetDMinusE;
+                        if (DMinusE.size() == 1) {
+                            indexOfSubsetDMinusE = *DMinusE.begin();
+                        } else {
+                            indexOfSubsetDMinusE = fetchIndexofMapofSets(DMinusE, subsetMap);
+                        }
+
+                        if ((distanceEJ + MATRIX(dp_cache, indexOfSubsetDMinusE, j)) < distance1) {
+                            distance1 = distanceEJ + (MATRIX(dp_cache, indexOfSubsetDMinusE, j));
+                        }
+                    }
+                }
+
+                for (igraph_integer_t k = 0; k < no_of_nodes; k++) {
+                    MATRIX(dp_cache, indexOfSubsetD, k) = std::min(MATRIX(dp_cache, indexOfSubsetD, k), MATRIX(distance, k, j) + distance1);
+                }
+            }
+        }
+    }
+
+    igraph_real_t distance2 = IGRAPH_INFINITY;
+
+    for (igraph_integer_t j = 0; j < no_of_nodes; j++) {
+        igraph_real_t distance1 = IGRAPH_INFINITY;
+        IGRAPH_ALLOW_INTERRUPTION();
+        for (igraph_integer_t subset_C_iterator = 0; subset_C_iterator < no_of_terminals; subset_C_iterator++) {
+            igraph_integer_t F = VECTOR(steiner_terminals_copy)[subset_C_iterator];
+            igraph_integer_t distanceFJ = MATRIX(distance, F, j);
+
+            int_set CMinusF;
+
+            for (igraph_integer_t k = 0; k < no_of_terminals; k++) {
+
+                if (VECTOR(steiner_terminals_copy)[k] != F) {
+                    CMinusF.insert(VECTOR(steiner_terminals_copy)[k]);
+                }
+            }
+
+            igraph_integer_t indexOfSubsetCMinusF = fetchIndexofMapofSets(CMinusF, subsetMap);
+
+            if (distanceFJ != 0 && (distanceFJ + (MATRIX(dp_cache, indexOfSubsetCMinusF, j)) < distance1)) {
+                distance1 = distanceFJ + (MATRIX(dp_cache, indexOfSubsetCMinusF, j));
+            }
+        }
+
+        if (q != j && MATRIX(distance, q, j) + distance1 < distance2) {
+            distance2 = MATRIX(distance, q, j) + distance1;
+        }
+    }
+    *res = distance2;
+
+    int_set newSet;
+    for (igraph_integer_t i = 0; i < igraph_vector_int_size(&steiner_terminals_copy); i++) {
+        newSet.insert(VECTOR(steiner_terminals_copy)[i]);
+    }
+    igraph_integer_t indexD = fetchIndexofMapofSets(newSet, subsetMap);
+
+    igraph_vector_int_t vectorlist_all;
+    IGRAPH_VECTOR_INT_INIT_FINALLY(&vectorlist_all, 0);
```

Comment:
This is a list of vertices, not a list of vectors, so it should have been named accordingly. However, it does not appear to be used at all. If not needed, please remove it.

Reply:
This is used to store the result which are vertices in the Steiner tree and is to be used as output(Output is length of Steiner tree and the tree itself). We are changing the name as you've suggested.

Reply:
The vertices of the tree are not at the moment returned by the top-level function. First, we need to decide whether it makes sense to return them. Currently, my opinion is that this is not necessary, as the vertices (as opposed to edges) would be rarely needed. If they are needed for some special application, they can be computed from the edges. Do you agree?

If the vertices are not returned or used anyway, then there is no need to record them, so this variable can be removed.

Reply:
Understood. I have removed vectorlist_all from as input parameter to the function calculating tree and it's references inside the function